infrastructure/src/main/java/org/corfudb/infrastructure/configuration/ServerConfiguration.java

```diff
@@ -0,0 +1,722 @@
+        mapping.put("--HandshakeTimeout", HANDSHAKE_TIMEOUT);
+        mapping.put("--cluster-id", CLUSTER_ID);
+        mapping.put("--tls-ciphers", TLS_CIPHERS);
+        mapping.put("--tls-protocols", TLS_PROTOCOLS);
+        mapping.put("--metrics", ENABLE_METRICS);
+        mapping.put("--snapshot-batch", SNAPSHOT_BATCH_SIZE);
+        mapping.put("--lock-lease", LOCK_LEASE_DURATION);
+        mapping.put("--Prefix", THREAD_PREFIX);
+        mapping.put("--metadata-retention", METADATA_RETENTION);
+        mapping.put("<port>", SERVER_PORT);
+
+        return mapping;
+    }
+
+
+    public ServerConfiguration setServerDirectory(String path) {
+        setProperty(SERVER_DIR, path);
+        return this;
+    }
+
+    public String getServerDir() {
+        return getString(SERVER_DIR);
+    }
+
+    public String getLogDir() {
+        return getServerDir() + File.separator + "log";
+    }
+
+    public ServerConfiguration setSingleMode(boolean enable) {
+        setProperty(SINGLE_MODE, enable);
+        return this;
+    }
+
+    public boolean isSingleMode() {
+        return getBoolean(SINGLE_MODE);
+    }
+
+    public ServerConfiguration setClusterId(String clusterId) {
+        setProperty(CLUSTER_ID, clusterId);
+        return this;
+    }
+
+    public String getClusterId() {
+        return getString(CLUSTER_ID, DEFAULT_CLUSTER_ID);
+    }
+
+    public ServerConfiguration setNumIOThreads(int num) {
+        setProperty(NUM_IO_THREADS, num);
+        return this;
+    }
+
+    public int getNumIOThreads() {
+        return getInt(NUM_IO_THREADS, Runtime.getRuntime().availableProcessors());
+    }
+
+    public ServerConfiguration setHostAddress(String address) {
+        setProperty(HOST_ADDRESS, address);
+        return this;
+    }
+
+    public String getHostAddress() {
+        return getString(HOST_ADDRESS);
+    }
+
+    public ServerConfiguration setServerPort(int port) {
+        setProperty(SERVER_PORT, port);
+        return this;
+    }
+
+    public String getLocalServerEndpoint() {
+        return getHostAddress() + ":" + getServerPort();
+    }
+
+    public int getServerPort() {
+        return getInt(SERVER_PORT);
+    }
+
+    public ServerConfiguration setNetworkInterface(String networkInterface) {
+        setProperty(NETWORK_INTERFACE, networkInterface);
+        return this;
+    }
+
+    public String getNetworkInterface() {
+        return getString(NETWORK_INTERFACE);
+    }
+
+    public ServerConfiguration setHandshakeTimeout(int timeout) {
+        setProperty(HANDSHAKE_TIMEOUT, timeout);
+        return this;
+    }
+
+    public int getHandshakeTimeout() {
+        return getInt(HANDSHAKE_TIMEOUT, Integer.parseInt(DEFAULT_HANDSHAKE_TIMEOUT));
+    }
+
+    public ServerConfiguration setMetadataRetention(int numFiles) {
+        if (numFiles < 1) {
+            throw new IllegalArgumentException("Max number of metadata files to retain must be greater than 0.");
+        }
```

Comment:
Is it worth adding a similar check in other places in this class where we expect integer parameters to satisfy certain constraints?

Reply:
I believe that the values of these properties are checked in their usages.